api/graphql/tests/test_reservation_units/test_reservation_units_query_opening_hours.py

```diff
@@ -0,0 +1,274 @@
+import datetime
+from zoneinfo import ZoneInfo
+
+import pytest
+from django.utils.timezone import get_default_timezone
+
+from reservation_units.models import ReservationUnit
+from tests.factories import (
+    OriginHaukiResourceFactory,
+    ReservableTimeSpanFactory,
+    ReservationUnitFactory,
+)
+
+# Applied to all tests
+pytestmark = [
+    pytest.mark.django_db,
+]
+
+DEFAULT_TIMEZONE = get_default_timezone()
+
+UTCOFFSET = f"0{int(datetime.datetime(2021, 5, 1, tzinfo=DEFAULT_TIMEZONE).utcoffset().total_seconds()/3600)}:00"
+
+
+@pytest.fixture()
+def reservation_unit() -> ReservationUnit:
+    origin_hauki_resource = OriginHaukiResourceFactory(
+        id="999",
+        opening_hours_hash="test_hash",
+        latest_fetched_date="2021-05-03",
+    )
+    return ReservationUnitFactory(origin_hauki_resource=origin_hauki_resource)
+
+
+def _get_date(*, day: int = 1, hour: int = 0, tzinfo: ZoneInfo | None = None):
+    return datetime.datetime(2023, 5, day, hour, 0, 0, tzinfo=tzinfo or DEFAULT_TIMEZONE)
+
+
+GET_QUERY = """
+    query {
+        reservationUnitByPk(pk: %i) {
+            reservableTimeSpans(startDate:"2023-05-01" endDate:"2024-01-01"){
+                startDatetime endDatetime
+            }
+        }
+    }
+"""
+
+
+def test__gql__reservation_unit__reservable_time_spans__no_origin_hauki_resource(graphql, reservation_unit):
+    reservation_unit.origin_hauki_resource = None
+    reservation_unit.save()
+
+    response = graphql(GET_QUERY % reservation_unit.id)
+
+    assert response.get("errors") is None
+    assert response.data == {
+        "reservationUnitByPk": {
+            "reservableTimeSpans": None,
+        }
+    }
+
+
+def test__gql__reservation_unit__reservable_time_spans__no_start_date(graphql, reservation_unit):
+    reservation_unit.origin_hauki_resource = None
+    reservation_unit.save()
+
+    response = graphql(
+        """
+        query {
+            reservationUnitByPk(pk: %i) {
+                reservableTimeSpans(endDate:"2024-01-01"){
+                    startDatetime endDatetime
+                }
+            }
+        }
+        """
+        % reservation_unit.id
+    )
+
+    assert response.get("errors") is None
+    assert response.data == {
+        "reservationUnitByPk": {
+            "reservableTimeSpans": None,
```

Comment:
Not sure this is what I'd expect. Seem like you'd want to enforce that both `startDate` and `endDate` are required in the GraphQL schema, and give an error back if they are missing. Right now, they are typed as `Date`, which means they aren't required (Should be `Date!` if required). I think you can add `required=True`.